app/exporters/base_exporter.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+class BaseExporter
+  DEFAULT_ATTRIBUTE_NAMES = %w[id created_at updated_at].freeze
+
+  def initialize(records, attribute_names)
+    @records = records
+    @attribute_names = attribute_names
+  end
+
+  def to_csv
+    return "" if @records.empty?
+
+    model = @records.first.class
+    data = process_data(attributes)
+
+    CSV.generate(headers: true) do |csv|
+      csv << attributes.map { |attr| model.human_attribute_name(attr) }
+
+      data.each do |record|
+        csv << record.values
+      end
+    end
+  end
+
+  class UndefinedAttributeExporter < StandardError
+    def initialize(attribute, model, exporter)
+      super("Attribute '#{attribute}' is not defined by either #{model} or #{exporter}")
+    end
+  end
+
+  private
+
+  def process_data(attribute_names)
+    @records.map do |record|
+      attribute_names.index_with do |attribute_name|
+        record_attribute_value(record, attribute_name)
+      end
+    end
+  end
+
+  def attributes
+    @attributes ||= (self.class::DEFAULT_ATTRIBUTE_NAMES + @attribute_names).uniq
+  end
+
+  def record_attribute_value(record, attribute_name)
+    if respond_to? attribute_name
```

Comment:
```suggestion
    if respond_to?(attribute_name)
```

It's better to use brackets on condition to avoid issues when adding more elements